### ISSUE TITLE
TypeError: Cannot read property 'taxes_id' of undefined

### ISSUE THREAD

**Issue by mawke2020** [odoo/odoo#64823]
Impacted versions:

Steps to reproduce:

Current behavior:

Expected behavior:

Video/Screenshot link (optional)
[error 2021-01-20-08-31-44.txt](https://github.com/odoo/odoo/files/5844000/error.2021-01-20-08-31-44.txt)
:

Support ticket number submitted via odoo.com/help (optional):

**Comment by pedrobaeza** [comment]
Hi, thanks for reporting. Are you able to reproduce the problem in runbot infrastructure (http://runbot.odoo.com/runbot - click on the blue door icon of the version row - admin/admin)?

If so, please explain with all the possible detailed steps to get to the error. Not having them, it's very difficult to hunt down the problem. Please don't be shy of adding details, and even better if you provide some screenshots or videos.

If not, this can be due to a configuration change, a custom/community module you have deployed in your DB or an outdated code. Please report back if it's the case.

For keeping a clean house in this issue tracker, if you don't answer in several days, this will be closed assuming it was one of the stated things.

**Comment by mawke2020** [comment]
hi   thanks for your replay

still im facing same problem when im using pos sale odoo 14  the TypeError: Cannot read property 'taxes_id' of undefined


https://user-images.githubusercontent.com/75810086/105321393-406eaf00-5b7c-11eb-938e-b1621a892251.mp4

[error 2021-01-20-08-31-44.txt](https://github.com/odoo/odoo/files/5847709/error.2021-01-20-08-31-44.txt)

**Comment by pedrobaeza** [comment]
Unless you reproduce it on runbot and give detailed steps that leads to it, this can't be considered as an Odoo issue, but your end's one.

**Comment by tingasss** [comment]
Hi @mawke2020,I currently  have the same  problem!  you were able to solve the problem?

**Comment by pbuitronf** [comment]
@tingasss @mawke2020  I'm having the same issue. Were you able to fix it? Thanks

**Comment by mhabboush97** [comment]
@pedrobaeza this issue is reproducible on runbot
to reproduce archive a product that is sold in a pos order, then open the order management from pos ui
The issue is that get_all_prices function tried to read taxes from product, but product is undefined since it is archived
this behavior might also happen in orderline model initialization process. but no errors are thrown there

**Comment by mhabboush97** [comment]
@pedrobaeza this issue is reproducible on runbot
to reproduce archive a product that is sold in a pos order, then open the order management from pos ui
The issue is that get_all_prices function tried to read taxes from product, but product is undefined since it is archived
this behavior might also happen in orderline model initialization process. but no errors are thrown there

a side note, the way PoS process order history is buggy, it re-creates orders from data, so if you sold a product with 15% tax, then after a while you changed its tax to 10%, the order that have the product with 15% tax will appear as if it was sold with 10% tax, all fields will be incorrect, product price, order taxes & order total @caburj

**Comment by pedrobaeza** [comment]
OK, now with that steps it seems we have a reproducible path. @pimodoo maybe you can check this?

**Comment by jeje1307** [comment]
I also got the same error on Odoo 14 POS, any news ?

**Comment by mhabboush97** [comment]
@jeje1307 this issue happens when you have archived/deleted product that is in a pos order with pos order management enabled
A simple workaround is to modify the function 'get_all_prices'. At the beginning of the function, check if product is undefined, and is so just return immediately

**Comment by jeje1307** [comment]
@mhabboush97 thank you for the information.

Anyone have a procedure (i'm New to odoo) where i can do that in Windows 10 ? Which file or is it in the software ?

Sorry for the inconvenient, and thank you very much for your help.

**Comment by mhabboush97** [comment]
@jeje1307 a very naïve and quick solution is to navigate here https://github.com/odoo/odoo/blob/b16413a5d2072efcae619f968ea9f772f6654efa/addons/point_of_sale/static/src/js/models.js#L2278
in your installation files, and add this the above line
if (!product )
return

* I don't get why there is no follow-up from Odoo devs on such important issue

**Comment by jeje1307** [comment]
@mhabboush97 Thank you so much, it worked like a charm.

**Comment by mhabboush97** [comment]
@pedrobaeza @pimodoo a similar error also happens when you previously sold a product with tracking enabled, then disabled tracking and opened order management. It happens because init from json will try to append the tracking lines to the order line, but will throw error because the product doesn't have pack_lot_lines (because currently tracking is set to none)

**Comment by pedrobaeza** [comment]
Well, you mustn't do such things and don't remove tracking when the product has been used.

**Comment by mhabboush97** [comment]
@pedrobaeza not really, maybe if a product when out of stock, and before restocking it I don't want to track it anymore, happens?
but that's just a side effect of a bigger problem, creating order lines from products themselves

**Comment by pedrobaeza** [comment]
You shouldn't do that, even if no current stock in physical locations, as there's still quants in other virtual locations.

**Comment by mhabboush97** [comment]
@jeje1307 this happens when you leave active order in a pos session, close the session, archive (other actions might trigger the error as well?) a product that is in an orderline in that active order, open the session again.

@pedrobaeza another issue to fix I think, right?

**Comment by pedrobaeza** [comment]
Although it's true that you can crash the system doing such things, adding a lot of constraints of don't things adds a code complexity that it's not too much rewarded due to all the possible failing conditions. For me, it's common sense to not do certain things that involve opened PoS sessions.

**Comment by mhabboush97** [comment]
@pedrobaeza 
I recently encountered this issue when I leave product X in active PoS order and close the session, untick available_in_pos for product X, open the session again. Session doesn't open for same error in this ticket

**Comment by pedrobaeza** [comment]
For me it's the same: you shouldn't do such thing or expect side effects. Adding constraints for every possible user action is barely impossible and performance catastrophic. Just add to your knowledge that you can't do such thing, which is by logic.

I still let if Odoo team thinks this constraint still worths to be added even if we lose some performance.

**Comment by mhabboush97** [comment]
@pedrobaeza what do you mean by such a thing? removing product from PoS? changing available pricelists when session is already closed? changing product taxes? changing product tracking? 
All those actions either directly or indirectly stops PoS from working
Not to mention what I mentioned in https://github.com/odoo/odoo/issues/64823#issuecomment-790606185
and I am developer, so I can fix those issues for my clients, I am just notifying the Odoo team for some of the issues I find so they can provide better experience for end-users with no development background, but just wondering why little interest in fixing them

**Comment by Yenthe666** [comment]
Debatable @pedrobaeza, Odoo POS does break easily on this kind of stuff.
Let's ask @pimodoo

**Comment by cameleonRuoi** [comment]
it happens to me now, i just clear cache,
 and we don't use taxes so i set all sale taxe to false.

**Comment by mamiu** [comment]
Is there a solution now?
I don't think it's [that one](https://github.com/odoo/odoo/issues/64823#issuecomment-793061675) although it works.

@pimodoo is assigned to this issue but hasn't commented on it yet.
I'm keen to hear his assessment.

**Comment by framirez07** [comment]
Clearing the browser cache worked for me.

**Comment by krutin123** [comment]
> @tingasss @mawke2020 I'm having the same issue. Were you able to fix it? Thanks

need to remove cache and cookies from browser
